Fix for #601

jcdevil opened the pull request:
Fix to install on CentOS (tested with CentOS 7.7)

p commented on setup.py:
```diff
@@ -372,16 +372,17 @@ def detect_ssl_lib_from_libcurl_dll(self, libcurl_dll_path):
+            dist_name, dist_version, _ = platform.dist()
+            dist_version = dist_version.split('.')[0]
+            if dist_name != 'centos' or dist_version != '6':
+                return False
```
Since this check is now nested in the `hasattr` check, what I believe will happen is OSes other than centos will end up running the subsequent `find_library` code. Instead what should happen is the `find_library` code should only be executed on OSes that have `dist` and whose `dist` is `centos`.

jcdevil replied:
yep indeed => PR updated

jcdevil replied:
However, no test was failing => any new test to add ?!

p replied:
I think this is not necessary.

jcdevil commented:
Would it be possible to generate a new release with that patch to fix my current installation issue please?
Thx

Jean-Charles Deville


> On 17 Jan 2020, at 18:18, Oleg Pudeyev <notifications@github.com> wrote:
> 
> @p commented on this pull request.
> 
> Thank you for the patch.
> 
> In setup.py <https://github.com/pycurl/pycurl/pull/610#discussion_r368046926>:
> 
> >          import platform
>          from ctypes.util import find_library
>          os_name = platform.system()
>          if os_name != 'Linux':
>              return False
> -        dist_name, dist_version, _ = platform.dist()
> -        dist_version = dist_version.split('.')[0]
> -        if dist_name != 'centos' or dist_version != '6':
> -            return False
> +        if hasattr(platform, 'dist'):
> +            dist_name, dist_version, _ = platform.dist()
> +            dist_version = dist_version.split('.')[0]
> +            if dist_name != 'centos' or dist_version != '6':
> +                return False
> Since this check is now nested in the hasattr check, what I believe will happen is OSes other than centos will end up running the subsequent find_library code. Instead what should happen is the find_library code should only be executed on OSes that have dist and whose dist is centos.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/pycurl/pycurl/pull/610?email_source=notifications&email_token=AAUDI6G5SI2CVEIFTPKLFKLQ6HR7TA5CNFSM4KILPFP2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCSGBMMA#pullrequestreview-344725040>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAUDI6HNYPRWMIJQ5OIGIMDQ6HR7TANCNFSM4KILPFPQ>.
>
